test/integration/master/master_test.go

```diff
@@ -195,6 +253,59 @@ func TestAutoscalingGroupBackwardCompatibility(t *testing.T) {
 	}
 }
 
+func TestAppsGroupBackwardCompatibility(t *testing.T) {
+	_, s := framework.RunAMaster(nil)
+	defer s.Close()
+	transport := http.DefaultTransport
+
+	requests := []struct {
+		verb                string
+		URL                 string
+		body                string
+		expectedStatusCodes map[int]bool
+		expectedVersion     string
+	}{
+		// Post to extensions endpoint and get back from both: extensions and apps
+		{"POST", extensionsPath("deployments", metav1.NamespaceDefault, ""), deploymentExtensions, integration.Code201, ""},
+		{"GET", extensionsPath("deployments", metav1.NamespaceDefault, "test-deployment"), "", integration.Code200, testapi.Extensions.GroupVersion().String()},
+		{"GET", appsPath("deployments", metav1.NamespaceDefault, "test-deployment"), "", integration.Code200, testapi.Apps.GroupVersion().String()},
+		{"DELETE", extensionsPath("deployments", metav1.NamespaceDefault, "test-deployment"), "", integration.Code200, api.Registry.GroupOrDie(v1.GroupName).GroupVersion.String()},
```

Comment:
I don't think this is needed. The 2nd POST is happening after DELETE, and since both endpoints share the same storage it should fail POST if the previous deletion failed, assuming I'm using the same name for both deployments, and as you can see, I am..